pkg/debug/checkups/tuf.go

```diff
@@ -0,0 +1,104 @@
+package checkups
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"runtime"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/autoupdate/tuf"
+)
+
+type (
+	tufCheckup struct {
+		k       types.Knapsack
+		status  Status
+		summary string
+		data    map[string]any
+	}
+
+	customTufRelease struct {
+		Custom struct {
+			Target string `json:"target"`
+		} `json:"custom"`
+	}
+
+	tufRelease struct {
+		Signed struct {
+			Targets map[string]customTufRelease `json:"targets"`
+		} `json:"signed"`
+	}
+)
+
+func (tc *tufCheckup) Data() map[string]any  { return tc.data }
+func (tc *tufCheckup) ExtraFileName() string { return "" }
+func (tc *tufCheckup) Name() string          { return "Tuf Version" }
+func (tc *tufCheckup) Status() Status        { return tc.status }
+func (tc *tufCheckup) Summary() string       { return tc.summary }
+
+func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
+	tc.data = make(map[string]any)
+	if !tc.k.Autoupdate() {
+		return nil
+	}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+
+	tufEndpoint := fmt.Sprintf("%s/repository/targets.json", tc.k.TufServerURL())
+	tufUrl, err := parseUrl(tc.k, tufEndpoint)
+	if err != nil {
+		return err
+	}
+
+	response, err := tc.fetchTufVersion(httpClient, tufUrl)
```

Comment:
I've seen this in a couple of spots -- havingt a helper for the fetch makes sense, but why generate the URL up here, and not in the same helper?